.github/workflows/template-build.yml

```diff
@@ -0,0 +1,57 @@
+on:
+  workflow_call:
+    inputs:
+      release-tag-name:
+        required: true
+        type: string
+      push:
+        required: true
+        type: boolean
+env:
+  imageName: "rancher/harvester-seeder"
+
+jobs:
+  dapper-build:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      id-token: write
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Run dapper
+        run: make ci
+
+      - name: Read some Secrets
+        uses: rancher-eio/read-vault-secrets@main
+        if: ${{ inputs.push == true }}
+        with:
+          secrets: |
+            secret/data/github/repo/${{ github.repository }}/dockerhub/rancher/credentials username | DOCKER_USERNAME ;
+            secret/data/github/repo/${{ github.repository }}/dockerhub/rancher/credentials password | DOCKER_PASSWORD
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v3
+        if: ${{ inputs.push == true }}
+        with:
+          username: ${{ env.DOCKER_USERNAME }}
+          password: ${{ env.DOCKER_PASSWORD }}
+
+      - name: Docker Build
+        uses: docker/build-push-action@v5
+        with:
+          provenance: false
+          context: .
+          platforms: linux/amd64,linux/arm64
+          file: package/Dockerfile
+          push: ${{ inputs.push }}
+          tags: ${{env.imageName}}:${{ inputs.release-tag-name }}
+          build-args: |
+            VERSION=${{ github.ref_name }}-${{ github.sha }}-head
```

Comment:
Is this version need to follow `release-tag-name`, so we don't get `<branch>-<sha>-head` for a tag version?

Reply:
I realized VERSION is not used in Dockerfile as build args, maybe we can remove it?

Reply:
good spot.. removed it.

the branch sha based tags are not pushed to dockerhub so i assume that is immaterial.